python/simulation.py

```diff
@@ -1633,22 +1632,36 @@ def _evaluate_dft_objects(self):
             if dft.swigobj is None:
                 dft.swigobj = dft.func(*dft.args)
 
-    def add_dft_fields(self, components, freq_min, freq_max, nfreq, where=None, center=None, size=None, yee_grid=False):
+    def add_dft_fields(self, *args, **kwargs):
+        components = args[0]
+        if len(args) == 4 and isinstance(args[1],(int,float)) and isinstance(args[2],(int,float)) and isinstance(args[3],int):
+            fcen = args[1]
+            df = args[2]
+            nfreq = args[3]
+            freq = [fcen] if nfreq == 1 else np.linspace(fcen-0.5*df,fcen+0.5*df,nfreq)
+        elif len(args) == 2 and isinstance(args[1],(np.ndarray,list)):
+            freq = args[1]
+        else:
+            raise TypeError("add_dft_fields only accepts fcen,df,nfreq (3 numbers) or freq (array/list)")
```

Comment:
Can we please use my `fix_dft_args` suggestion so that we don't have these same 9 lines copy-and-pasted in to many different functions?

Reply:
Done.